src/lib.rs

```diff
@@ -29,7 +29,30 @@
 //! API changed significantly since then.
 
 #![cfg_attr(not(feature = "std"), no_std)]
-#![allow(clippy::trivially_copy_pass_by_ref)] // for compatibility with methods on char and u8
+// Clippy lints
+#![warn(
+    clippy::pedantic,
```

Comment:
I'm not sure if I really like this, but I think we should try using it. Maybe I also get used to it and like the pedantic lints in the end :)